### DISCUSSION TITLE
Question about the log system

### DISCUSSION THREAD

**Discussion by exiawsh** [Lightning-AI/pytorch-lightning#11670] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
Hello, I have some question about the self.log function and batch_size during the trainer.
If I have two GPUs, and I want to train my model with batch_size 16 per GPU and I use DDP, so what's the number of batch_size in Datamodule and what's the number of batch_size in self.log, If I want to calculate my metrics correctly?

**Comment by rohitgr7** [comment, marked as answer]
hey @exiawsh 

it should stay as batch_size for a single device only. With DDP if you set `batch_size=7`, then each device gets the batch of `batch_size=7`, and effective batch_size increases with the number of devices. Now if you want to log by accumulating metrics across devices, you need to set `sync_dist=True`. Check out the section here: https://pytorch-lightning.readthedocs.io/en/latest/extensions/logging.html#automatic-logging

**Reply by exiawsh** [reply]
Thank you for your answer. If I just use on_epoch = True, shall I still use sync_dist=True?

**Reply by rohitgr7** [reply]
yes. on_epoch just ensures accumulation is done over all the batches on the device, not across devices :)

**Reply by rohitgr7** [reply]
since sync_dist uses all_reduce operations which is expensive, it is not enabled until users want it.

**Reply by exiawsh** [reply]
okok, I got it. Thank you!
